packages/hash/frontend/src/components/hooks/useUpdateAuthenticatedUser.ts

```diff
@@ -0,0 +1,117 @@
+
+type UpdateAuthenticatedUserParams = {
+  shortname?: string;
+  preferredName?: string;
+};
+
+export const useUpdateAuthenticatedUser = () => {
+  const { authenticatedUser, refetch } = useAuthenticatedUser();
+
+  const [updatePersistedEntity] = useMutation<
+    UpdatePersistedEntityMutation,
+    UpdatePersistedEntityMutationVariables
+  >(updatePersistedEntityMutation, { errorPolicy: "all" });
+
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const updateAuthenticatedUser = useCallback(
+    async (
+      params: UpdateAuthenticatedUserParams,
+    ): Promise<{
+      updatedAuthenticatedUser?: AuthenticatedUser;
+      errors?: readonly GraphQLError[] | undefined;
+    }> => {
+      if (!params.shortname && !params.preferredName) {
+        return { updatedAuthenticatedUser: authenticatedUser };
+      }
+
+      const [
+        {
+          data: { me: latestSubgraph },
+        },
+      ] = await refetch();
+
+      const entityId = latestSubgraph.roots[0]!;
+
+      /** @todo: use a partial update mutation instead */
+      const { properties: currentProperties } = mustGetEntity({
+        subgraph: latestSubgraph as unknown as Subgraph,
+        entityId,
+      });
+
+      const { errors } = await updatePersistedEntity({
+        variables: {
+          entityId,
+          updatedProperties: {
+            ...currentProperties,
+            ...(params.shortname
+              ? {
+                  [extractBaseUri(types.propertyType.shortName.propertyTypeId)]:
+                    params.shortname,
+                }
+              : {}),
+            ...(params.preferredName
+              ? {
+                  [extractBaseUri(
+                    types.propertyType.preferredName.propertyTypeId,
+                  )]: params.preferredName,
+                }
+              : {}),
+          },
+        },
+      });
+
+      if (errors && errors.length > 0) {
+        return { errors };
+      }
+
+      const [
+        {
+          data: { me: updatedSubgraph },
+        },
+        kratosSession,
+      ] = await refetch();
+
+      const updatedAuthenticatedUser = constructAuthenticatedUser({
+        userEntityId: updatedSubgraph.roots[0]!,
+        /**
+         * @todo: ensure this subgraph contains the incoming links of orgs
+         * at depth 2 to support constructing the `members` of an `Org`.
+         *
+         * @see https://app.asana.com/0/1202805690238892/1203250435416412/f
+         */
+        subgraph: updatedSubgraph as unknown as Subgraph,
+        kratosSession: kratosSession!,
+      });
+
+      return { updatedAuthenticatedUser };
+    },
+    [authenticatedUser, refetch, updatePersistedEntity],
+  );
+
+  return {
+    updateAuthenticatedUser: async (params: UpdateAuthenticatedUserParams) => {
+      try {
+        setLoading(true);
+        return updateAuthenticatedUser(params);
+      } finally {
+        setLoading(false);
+      }
+    },
+    loading,
+  };
```

Comment:
The pattern we usually follow for mutation hooks is returning an array with the update function and an object with the loading variable. 

```suggestion
  return [
    async (params: UpdateAuthenticatedUserParams) => {
      try {
        setLoading(true);
        return updateAuthenticatedUser(params);
      } finally {
        setLoading(false);
      }
    },
    { loading },
  ] as const;
```

Reply:
Good spot - thanks for letting me know :)

Have addressed in https://github.com/hashintel/hash/pull/1299/commits/df064bd05499fc7b6a909b4ac9fece3857c5f503